backend/apps/ifc_validation_bff/views_legacy.py

```diff
@@ -100,50 +100,48 @@ def file_contains_string(file_name, fragment):
 
 @functools.lru_cache(maxsize=1024)
 def get_feature_url(feature_code, feature_version):
+    # TODO - fetch remote/sha based on code/version; for now we point to main and development repo (configurable)
+    file_folder = os.path.dirname(os.path.realpath(__file__))
+    rules_folder = os.path.join(file_folder, '../ifc_validation/checks/ifc_gherkin_rules/features/rules')
 
-    # TODO - fetch remote/sha based on code/version; for now we point to main repo (configurable)
+    version_tag = f'@version{feature_version}'
 
-    file_folder = os.path.dirname(os.path.realpath(__file__))
-    feature_folder = os.path.join(file_folder, '../ifc_validation/checks/ifc_gherkin_rules/features')
+    for root, _, files in os.walk(rules_folder):
+        for file in files:
+            file_name = os.fsdecode(file)
+            fq_file_name = os.path.join(root, file_name)
 
-    for file in os.listdir(feature_folder):
-        file_name = os.fsdecode(file)
-        fq_file_name = os.path.join(feature_folder, file_name)
-        version_tag = f'@version{feature_version}'
-        
-        if file_name.endswith(".feature") and file_name.startswith(feature_code) and file_contains_string(fq_file_name, version_tag):
-            return FEATURE_URL + file_name
+            if file_name.endswith(".feature") and file_name.startswith(feature_code) and file_contains_string(fq_file_name, version_tag):
+                return os.path.join(FEATURE_URL, feature_code[:3] , file_name) 
     
     return None
 
 
 @functools.lru_cache(maxsize=1024)
 def get_feature_description(feature_code, feature_version):
-
-    # TODO - could probably use some of the ifc_gherkin_rules methods here?
-
     file_folder = os.path.dirname(os.path.realpath(__file__))
-    feature_folder = os.path.join(file_folder, '../ifc_validation/checks/ifc_gherkin_rules/features')
-    
-    for file in os.listdir(feature_folder):
-        file_name = os.fsdecode(file)
-        fq_file_name = os.path.join(feature_folder, file_name)
-        version_tag = f'@version{feature_version}'
+    rules_folder = os.path.join(file_folder, '../ifc_validation/checks/ifc_gherkin_rules/features/rules')
 
-        if file_name.endswith(".feature") and file_name.startswith(feature_code) and file_contains_string(fq_file_name, version_tag):
-            
-            gherkin_desc = ''
-            reading = False
+    version_tag = f'@version{feature_version}'
+
+    for root, _, files in os.walk(rules_folder):
+        for file in files:
+            file_name = os.fsdecode(file)
+            fq_file_name = os.path.join(root, file_name)
 
-            with open(os.path.join(feature_folder, file_name), 'r') as input:
+            if file_name.endswith(".feature") and file_name.startswith(feature_code) and file_contains_string(fq_file_name, version_tag):
```

Comment:
Please refactor out the common logic here into a separate function like get_feature_filename() that we can use to get the URL as well as description.